streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableFilter.java

```diff
@@ -17,32 +17,45 @@
 package org.apache.kafka.streams.kstream.internals;
 
 import org.apache.kafka.streams.kstream.Predicate;
+import org.apache.kafka.streams.kstream.StateStoreType;
+import org.apache.kafka.streams.kstream.Windowed;
 import org.apache.kafka.streams.processor.AbstractProcessor;
 import org.apache.kafka.streams.processor.Processor;
 import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.state.SessionStore;
 import org.apache.kafka.streams.state.TimestampedKeyValueStore;
+import org.apache.kafka.streams.state.TimestampedWindowStore;
 import org.apache.kafka.streams.state.ValueAndTimestamp;
 
 class KTableFilter<K, V> implements KTableProcessorSupplier<K, V, V> {
     private final KTableImpl<K, ?, V> parent;
-    private final Predicate<? super K, ? super V> predicate;
+    private final Predicate<?, ? super V> predicate;
     private final boolean filterNot;
     private final String queryableName;
     private boolean sendOldValues = false;
+    private final StateStoreType stateStoreType;
 
     KTableFilter(final KTableImpl<K, ?, V> parent,
-                 final Predicate<? super K, ? super V> predicate,
+                 final Predicate<?, ? super V> predicate,
                  final boolean filterNot,
-                 final String queryableName) {
+                 final String queryableName,
+                 StateStoreType stateStoreType) {
         this.parent = parent;
         this.predicate = predicate;
         this.filterNot = filterNot;
         this.queryableName = queryableName;
+        this.stateStoreType = stateStoreType;
     }
 
     @Override
-    public Processor<K, Change<V>> get() {
-        return new KTableFilterProcessor();
+    public Processor<?, Change<V>> get() {
```

Comment:
Use `KTableFilter` as an example. We need to return different processor upon the state store type for all the supplier get function calls. This is because the assumption of getting a key-value store is no longer valid.